### PR TITLE
Add testing helpers for alt stores

### PR DESCRIPTION
**Pull request by dphaener**
This is a naive first implementation. But it does work for basic
testing. I'd like to find a way to return 'this' instead of a promise
chain, so any help on this is appreciated.

### CODE REVIEW

**Comment by dphaener** [comment]
@zackify @goatslacker This is a naive first implementation. It *works*, but the one thing I'm not happy about is returning a promise chain. I'd like to be able to return `this` so we can continue to chain methods on like the original test suite.

**Comment by zackify** [comment]
I haven't thought this through but maybe calling a function inside of the setState that is an async- await so that you can wait on it then just callback to the function

**Comment by zackify** [comment]
Also, I don't really want anything other than jsdom as a direct dependency if possible

**Comment by dphaener** [comment]
@zackify @goatslacker I really like this new implementation. It now allows all the functions to be chainable, and obfuscates the `.wait` function to a behind the scenes player. 

Thanks to @zackify for the inspiration.

**Comment by dphaener** [comment]
@zackify I'll add some docs as soon as you think it looks good.

**Comment by dphaener** [comment]
@zackify @goatslacker Added another function `.call` that allows you to "synchronously" call a function and then continue to chain function calls.

**Comment by zackify** [comment]
Looks pretty cool. I think we could make .call actually hidden and make .addTodo something that gets added into this so it's even more awesome to use. Awesome work. Before merging though, if alt is a dev dependency how will it be in imported if it's not installed? I don't know a good way around this. Do you?

**Comment by dphaener** [comment]
No, I can't see a good way. Like I said, we have to assume that if someone is trying to test an Alt store, that it will be available for import. We can also be sure to include it in the docs.

**Comment by dphaener** [comment]
>> I think we could make .call actually hidden and make .addTodo something that gets added into this so it's even more awesome to use.

Not sure I follow..

**Comment by zackify** [comment]
Okay. I just wasn't sure if it would check the parent projects node modules
but it does, right?

**Comment by zackify** [comment]
Test(store).addTodo vs .call('addTodo')

**Comment by dphaener** [comment]
That's a good question.

**Comment by dphaener** [comment]
> Test(store).addTodo vs .call('addTodo')

Yeah, but methods like that are likely specific to each stores implementation. Are you talking about doing something like Ruby's method_missing? So that you can just call a function and we will try to find it?

**Comment by zackify** [comment]
If it does do that. Which I think it actually does, go ahead and merge when
you update the readme / wiki or whatever we want to use now.

**Comment by dphaener** [comment]
Alright, I'll link it locally and try it out on my local project

**Comment by zackify** [comment]
Object.assign(this, this.store) maybe that would even work? ;) obviously
naive but still.

**Comment by dphaener** [comment]
Let me take a ten minute crack at it and see if I can make it work. Because that would be super awesome.

**Comment by dphaener** [comment]
@zackify I think we're ready here. Finally.

File: README.md [hunk 1]
```diff
@@ -76,3 +76,18 @@ The `.test` function will be given the component instance and the helpers array.
 ~~~
 
 You can see more examples in the tests directory.
+
+##Testing Alt Stores
+
```
Comment (zackify) [inline README.md]: would this be `import Test from 'legit-tests/alt/store'`? Can you add that

**Comment by zackify** [comment]
Merge in master, there are some changes, like where files are outputted. Then it's good

**Comment by dphaener** [comment]
@zackify Done deal

**Comment by zackify** [comment]
@dphaener tests now :p

**Comment by dphaener** [comment]
@zackify Let's try this again

File: src/alt/store.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import 'harmony-reflect'
+
+class TestStore {
+  constructor(store, actions) {
+    this.store = store
+    this.actions = actions
+  }
+
+  action(funcName, data) {
```
Comment (zackify) [inline src/alt/store.js]: is this needed anymore?

Comment (dphaener) [inline src/alt/store.js]: Probably not, since we're delegating all actions in the proxy.

Comment (zackify) [inline src/alt/store.js]: cool, remove that, then should finally be good haha

**Comment by dphaener** [comment]
@zackify Annnnnnd again

**Pull request merged**